src/core/Akka/Actor/Props.cs

```diff
@@ -353,15 +323,15 @@ public override int GetHashCode()
             SupervisorStrategy supervisorStrategy = null) where TActor : ActorBase
         {
             if (factory.Body is UnaryExpression)
-                return new DynamicProps<TActor>(factory.Compile());
+                return new Props(new FactoryConsumer<TActor>(factory.Compile()), DefaultDeploy, typeof(TActor), NoArgs);
 
-            var newExpression = factory.Body.AsInstanceOf<NewExpression>();
-            if (newExpression == null)
-                throw new ArgumentException("The create function must be a 'new T (args)' expression");
+            var newExpression = factory.Body as NewExpression
+                ?? throw new ArgumentException("The create function must be a 'new T (args)' expression");
 
             var args = newExpression.GetArguments();
+            args = args.Length > 0 ? args : NoArgs;
```

Comment:
Consider, if you don't do the deed commented below about pulling ConstructorInfo:

```
var args = newExpression.Arguments > 0 ? newExpression.GetArguments() : NoArgs;
```

Reply:
That's a good idea